### PR TITLE
layout

### PR DESCRIPTION
**Pull request by codemaker4**
This is a standard for how re-usable logic creations should be laid out, specifically what shape it should have and how the interfaces should be marked.

### CODE REVIEW

File: standards/layout.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+This is the main logic component layout standard for the SMLC.
+
+## New global terminology
+These are new terms that are used in this standard that should be made global, which will happen in a seperate proposal in the terminology repo.
+- A **component** is a Scrap Mechanic creation intended to be re-used inside of other logic creations.
+
+## Terminology
+- An **interface** is a set of logic gates that form an input or output of a creation (and components, which are creations)
+- An **internal** gate is the opposite of an interface gate, so these are gates that are NOT supposed to be connected to external circuitry.
+- A **flatpak** formfactor means that the component is one block tall, and thus entirely flat.
```
Comment (redsti-github) [inline standards/layout.md]: typo: `flatpack` should be spellt with a "c"

File: standards/layout.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# = interface gate
+G = internal gate
+
+  GG G
+#GGG GG 
+#GGGGGG#
+```
+
+## Interface markings
+To make interfaces clearly visible, and ideally, make their function easy to read with just a glance at a component's documentation, gere are some standards for marking interface gates:
```
Comment (redsti-github) [inline standards/layout.md]: typo: `gere` -> `here`

**Review by redsti-github — CHANGES_REQUESTED** [review]
fix typos (see comments in file)

**Comment by redsti-github** [comment]
some topics that still need discussion/clarification (in my opinion):
- currently, internal gates are allowed to be grayscale, and orange is an output gate, however, the default gate colour is orange, which means that all internal gates have to be painted to greyscale. In my opinion, this isn't a problem, as long as it's stated more clearly in the standard.
- secondly, I was wondering about the binary inputs. Currently, the gates must point in the direction of the most significant bit. That is okay, but what about floating point numbers or numbers where the units bit (bit which is in the 1's place) isn't at the very end? (that is, some gates represent fractions - half, quarter, eighth and so on)

File: standards/layout.md (hunk 1 above)
Comment (codemaker4) [reply standards/layout.md]: thx

File: standards/layout.md (hunk 2 above)
Comment (codemaker4) [reply standards/layout.md]: thx

**Comment by codemaker4** [comment]
- hmm, I heared that some people actually like to keep their gates orange instead of painting them black, so leaving that as an internal color might be nice.
- Also, in the counter example, green = increment and red = decrement would be much more obvious, but is not allowed. Maybe we should find a different way to distinguish between inputs and outputs? Personally, I prefer to use gate pointing direction as IO indicator, and to mark the LSB I paint the LSB a slightly lighter variant than the rest of that interface. But in a discord chat, Niknal brought up that some people like to stack multiple interfaces on top of each other. If you would do that, you could not see gates pointing forwards or backwards. But this is not a problem for using gate orientation for MSB indication, because then you can just point sideways.
- About floating point, we will probably take something like the IEEE fp standard, and rephrase it so it makes sense to someone who plays SM and is not a computer scientist. In that case, we just define in that standard how to do the gate orientation.

**Comment by codemaker4** [comment]
Maybe we should just not restrict what colors can be inputs or outputs, and instead just standardize how you should write down what color is input/output and what it does

**Comment by codemaker4** [comment]
The consortium voted against this proposal, so changes have to be made.

**Comment by Foxtrot0055** [comment]
The general idea of this doesn't seem too bad.
The gates pointing towards the most significant bit isn't that bad of an idea but it can't easily be changed after building something (which might be annoying if you forget it) so I'd just make the least significant bit a lighter shade (and a thing I do sometimes is that I make another series of gates directly working with the interface a darker shade) e.g. 16Bit register input (green) the most significant bit (light green) select register to be written 3bit(dark green) write (very dark) just an idea.

**Comment by codemaker4** [comment]
Alright, I made some changes while in a discord voice call. The current version is a bit temproary, some sections are a stub.
One big point brought up by Endoddo is that for a basic standard that is more or less mandatory for our community to follow, we should avoid restricting _how_ creations are built, and instead focus on how to document stuff. Perhaps stop working on this standard for now, and release it later as an optional one, and in the meantime create a new standard for mandatory documentation...

**Comment by codemaker4** [comment]
So I have made #5 for documentation. I propose that a standard for how to physically build components should be delayed until that proposal is approved.

**Comment by codemaker4** [comment]
I am just going to close this for now.

**Pull request closed without merging**